BitwardenShared/UI/Auth/VaultUnlockSetup/VaultUnlockSetupState.swift

```diff
@@ -0,0 +1,87 @@
+// MARK: - VaultUnlockSetupState
+
+/// An object that defines the current state of a `VaultUnlockSetupView`.
+///
+struct VaultUnlockSetupState: Equatable {
+    // MARK: Types
+
+    /// An enumeration of the vault unlock methods.
+    ///
+    enum UnlockMethod: Int, Equatable, Identifiable {
+        /// Face ID is used to unlock the vault.
+        case faceID
+
+        /// The user's pin code is used to unlock the vault.
+        case pin
+
+        /// Touch ID is used to unlock the vault.
+        case touchID
+
+        /// The accessibility identifier for the UI toggle.
+        var accessibilityIdentifier: String {
+            switch self {
+            case .faceID, .touchID:
+                "UnlockWithBiometricsSwitch"
+            case .pin:
+                "UnlockWithPinSwitch"
+            }
+        }
+
+        /// A key path for getting/setting whether the unlock method is turned on in the state.
+        var keyPath: WritableKeyPath<VaultUnlockSetupState, Bool> {
```

Comment:
I'm curious what the advantage is of using key paths here?

Reply:
I'd say it avoids if/switch logic to get/set the unlock method so it's more direct to use.

Reply:
Yeah, exactly. You can implement both a get (`state[keyPath: unlockMethod.keyPath]`) and set (`state[keyPath: unlockMethod.keyPath] = newValue`) without having to implement the switch twice.